service/history/replication/raw_task_converter.go

```diff
@@ -68,6 +69,20 @@ type (
 		clientClusterName string, // Some task converter may use the client cluster name.
 		serializer serialization.Serializer,
 	) SourceTaskConverter
+
+	syncVersionedTransitionTaskConverter struct {
```

Comment:
~~nit: feel like this separate struct is not really necessary, but no strong opinion.~~ 

I guess I have ^ impression because today the SourceTaskConverter is not really doing it's job. I see how this sync task converter saves us from passing a lot of references around. We can refactor the entire source task converter later.